Docs: Improve style documentation and mark the cuppertino style as experimental (#3309)
slint-ui/slint: docs/reference/src/advanced/style.md

<!-- Copyright © SixtyFPS GmbH <info@slint.dev> ; SPDX-License-Identifier: MIT -->
# Selecting a Widget Style

Slint offers a variety of [built-in widgets](../language/widgets/widgets.md) which can be accessed from `"std-widgets.slint"`. You can modify the look of these widgets by choosing a style.

The styles available include:

> -   `fluent`, `fluent-light`, `fluent-dark`: These variants belong to the **Fluent** style, which is based on the [Fluent Design System](https://fluent2.microsoft.design/).

Author: Yes, that's probably right.
What's valid syntax for table i can use here?

> -   `fluent`, `fluent-light`, `fluent-dark`: These variants belong to the **Fluent** style, which is based on the [Fluent Design System](https://fluent2.microsoft.design/).

Reviewer: Markdown table should be fine AFAIK

-   `material`, `material-light`, `material-dark`: These variants are part of the **Material** style, which follows the [Material Design](https://m3.material.io).
-   `cupertino`, `cupertino-light`, `cupertino-dark`: The **Cupertino** variants emulate the style used by macOS applications. (Experimental)
-   `native`: The **Native** style mimics the controls native to the platform on which they are used. This includes specific support for the appearance of controls on macOS and Windows. This style requires Qt to be installed on your system.

> Some styles have a `-light` and `-dark` variant, in addition to their standard variant.
> The standard variant will switch between the dark or light variant based on your desktop settings.

Reviewer: I suggest to flip the order around:

```suggestion
By default, the styles automatically adapt to the system's dark or light color setting. Select a `-light` or `-dark` variant to override the system setting and always show either dark or light colors.
```

The widget style is determined at your project's compile time. The method to select a style depends on how you use Slint.

If no style is selected, `native` becomes the default if Qt is installed. If `native` isn't available, `fluent` is selected as the default.

## Selecting a Widget Style with Rust:

> You can choose the style before starting your compilation by setting the `SLINT_STYLE` variable to the name of your chosen style.

Reviewer: ```suggestion
You can select the style before starting your compilation by setting the `SLINT_STYLE` environment variable to the name of your chosen style.
```

> When using the `slint_build` API, utilize the [`slint_build::compile_with_config()`](https://docs.rs/slint-build/newest/slint_build/fn.compile_with_config.html) function.

Reviewer: ```suggestion
When using the `slint_build` API, call the [`slint_build::compile_with_config()`](https://docs.rs/slint-build/newest/slint_build/fn.compile_with_config.html) function.
```

> When using the `slint_interpeter` API, employ the [`slint_interpreter::ComponentCompiler::set_style()`](https://docs.rs/slint-interpreter/newest/slint_interpreter/struct.ComponentCompiler.html#method.set_style) function.

Reviewer: ```suggestion
When using the `slint_interpeter` API, call the [`slint_interpreter::ComponentCompiler::set_style()`](https://docs.rs/slint-interpreter/newest/slint_interpreter/struct.ComponentCompiler.html#method.set_style) function.
```

## Selecting a Widget Style when using C++

Define a `SLINT_STYLE` CMake cache variable to contain the style name as a string. This can be done, for instance, on the command line:

> cmake -DSLINT_STYLE="material" /path/to/source

Reviewer: I don't know how to do this with the inline suggestion, but I think this should be surrounded by a code fence backtick series :)

Author: You can use more backticks to escape: 
````suggestion
```sh
cmake -DSLINT_STYLE="material" /path/to/source
```
````

## Previewing Designs With `slint-viewer`

> Choose the style either by setting the `SLINT_STYLE` environment variable, or by passing the style name with the `--style` argument:

Reviewer: ```suggestion
Select the style either by setting the `SLINT_STYLE` environment variable, or by passing the style name with the `--style` argument:
```
